projects/button_adc/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void prv_adc_callback(const GpioAddress *address, void *context) {

```

Comment:
The name of this function is a little misleading - it's not triggered by the `adc` library, it interacts with it. (The `adc` library actually can call callbacks - look in adc.h if you're interested.) Usually an "ADC callback" is triggered by the `adc` library, a "soft timer callback" is triggered by the `soft_timer` library, etc. So I might name this callback after the circumstances where it's called, i.e. `prv_button_press_callback` or something.